movie_search/media.py

```diff
@@ -1,113 +1,265 @@
+            with_genres=list(genres),
+            sort_by=sort_options,
+            watch_region=watch_region,
+            with_watch_providers=list(providers),
+            with_people=person_id,
+        )
+
+
+class TVSeriesStrategy(MediaService):
+    def fetch_from_api(self, tmdb_id):
+        movie_data = tmdb_api_obj.get_data_from_endpoint(f"/tv/{tmdb_id}")
+        video_data = tmdb_api_obj.get_data_from_endpoint(f"/tv/{tmdb_id}/videos")
+
+    def store_data(self, data):
+        tv_data, video_data = data
+        # Store TV series data
+        genres = data.get("genres")
+        tv, created = TVSeries.objects.get_or_create(
+            tmdb_api_obj_id=tv_data.get("id", 0),
+            name=tv_data.get("name", ""),
+            backdrop_path=tv_data.get("backdrop_path", ""),
+            tagline=tv_data.get("tagline", ""),
+            vote_count=tv_data.get("vote_count", 0),
+            vote_average=tv_data.get("vote_average", 0),
+            popularity=tv_data.get("popularity", 0),
+            first_air_date=tv_data.get("first_air_date", ""),
+            number_of_episodes=tv_data.get("number_of_episodes", 0),
+            number_of_seasons=tv_data.get("number_of_seasons", 0),
+            production_company=tv_data.get("production_company", ""),
+            overview=tv_data.get("overview", ""),
+            homepage=tv_data.get("homepage", ""),
         )
-        movie_recommendations.append(recommendation)
-    return movie_recommendations
-
-def get_genres_from_discover(genre_names):
-    genres =  Genre.objects.filter(name__in=genre_names).values_list(
-        "genre_id", flat=True
-    )
-    return genres
-
-def get_providers_from_discover(watch_provider_names):
-    providers = Provider.objects.filter(name__in=watch_provider_names).values_list(
-        "provider_id", flat=True
-    )
-    return providers
-
-def get_movie_discover_data(
-     genres, person_id, sort_options, region, watch_region, providers, year
-):
-    return tmdb_api.get_data_from_endpoint(
-        "/discover/movie",
-        region=region,
-        primary_release_year=year,
-        with_genres=list(genres),
-        sort_by=sort_options,
-        watch_region=watch_region,
-        with_watch_providers=list(providers),
-        with_people=person_id,
-    )
+        if genres is not None:
+            tv_genres = store_genres(genres)
+
+        tv.genres.add(*tv_genres)
+
+        tv_recommendations = store_recommendations(tv.tmdb_id)
+        tv.recommendation.add(*tv_recommendations)
+
+        videos = self.store_videos(tv, video_data)
+
+        return tv, videos
+
+    def store_genres(self, genres):
+        tv_genres = []
+
+        for row in genres:
+            genre, created = TVSeriesGenre.objects.get_or_create(
+                name=row.get("name", ""),
+                genre_id=row.get("id", 0),
+            )
+            tv_genres.append(genre)
+
+        return tv_genres
+
+    def store_videos(self, tv_obj, video_data):
```

Comment:
can (should?) some of these additional methods also be abstracted in the abstract parent class?

Reply:
Yeah, I am working on that!